src/store/lib/useHostingStore.js

```diff
@@ -0,0 +1,234 @@
+      // source.onmessage = console.log;
+
+      source.addEventListener('sys', event => {
+        // console.log(event);
+        const sysEvent = JSON.parse(event.data);
+        const time = new Date(sysEvent.object.firstTimestamp).getTime();
+        if (time > startTime) {
+          console.log(sysEvent);
+          setSysEvents(previous => [...previous, sysEvent]);
+        }
+      });
+
+      source.addEventListener('new', event => {
+        // console.log(event);
+        const elapsed = (Date.now() - startTime) / 1000;
+        if (event.data !== '') {
+          setNewEvents(previous => [...previous, event.data]);
+        }
+        if (event.data === '1-b') {
+          setBootMessage('Compiling Hoon');
+          setBootProgress(0.2);
+          console.log(`${elapsed}: compiling Hoon`);
+        } else if (event.data === '1-c') {
+          setBootProgress(0.25);
+        } else if (event.data === '%arvo-assembly') {
+          setBootMessage('Compiling Urbit OS');
+          setBootProgress(0.3);
+        } else if (event.data.startsWith('[%vane %')) {
+          const vane = getVaneName(event.data.slice(8, 9));
+          setBootMessage(`Compiling vane: ${vane}`);
+          setBootProgress(0.3 + 0.05 * getVaneNumber(event.data.slice(8, 9)));
+        } else if (event.data === 'pier: boot complete') {
+          setBootMessage('Ship starting');
+          setBootProgress(0.85);
+        }
+      });
+
+      source.addEventListener('run', event => {
+        if (event.data === 'goad: recompiling all apps') {
+          setBootProgress(1.0);
+          setBootMessage('Ship Launched');
+          syncStatus();
+        }
+        if (event.data !== '') {
+          setRunEvents(previous => [...previous, event.data]);
+        }
+      });
+    } catch (error) {
+      setError(error);
+    }
+  }, [
+    setError,
+    resetEvents,
+    client,
+    pointCursor,
+    setBootMessage,
+    startTime,
+    syncStatus,
+  ]);
+
+  syncStatus = useCallback(async () => {
+    try {
+      if (Nothing.hasInstance(pointCursor)) {
+        return;
+      }
+      const patp = ob.patp(pointCursor.value);
+      setError(undefined);
+
+      resetEvents();
+
+      const response = await client.getShipsByPatp(patp);
+
+      if (response.status === 404) {
+        setStatus(STATE.MISSING);
+      } else if (!response.ok) {
+        setError(response.statusText);
+      } else {
+        const ship = await response.json();
+
+        if (ship.status === 'Pending') {
+          setStatus(STATE.PENDING);
+          getEvents();
+        } else if (ship.status === 'Running') {
+          setStatus(STATE.RUNNING);
+        } else {
+          setError(ship);
+        }
+      }
+    } catch (error) {
+      setError(error);
+    }
+  }, [client, getEvents, pointCursor, resetEvents, setError, setStatus]);
+
+  const hostedShipUrl = useMemo(
+    () =>
+      `http://${ob.patp(pointCursor.getOrElse('')).slice(1)}.liam.tlon.network`,
```

Comment:
We probably want to put the base URL here in some config file or w/e similar to how we treat the contract addresses.